games/BlockPuzzle.js

```diff
@@ -0,0 +1,478 @@
+0333333333333330
+0333333333333330
+0000000000000000` ]
+)
+
+setSolids([player, rock1, rock2, target, enemy])
+
+let level = 0;
+const levels = [
+  map`
+aaaaaaaaaaaa
+ap..b..b..aa
+a..b..b..baa
+a....b..b.aa
+a..b...b..aa
+a....b..b.aa
+a..b...b..aa
+a...b...b.aa
+a..b.e.b.taa
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaa
+ap.b.b..b.aa
+a..b....b.aa
+a....b.b..aa
+a..b....b.aa
+a....b..b.aa
+a..b...b..aa
+a...b..b..aa
+a..b.e.b.taa
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaa
+ap..b..b..aa
+a..b..b..baa
+a...b.bb.baa
+a..b..bbbbaa
+a..b..b..baa
+a..b..bbbbaa
+a..b..b..baa
+a..b.eb..taa
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaa
+a..b..b.b..a
+a..b..b..b.a
+ap.b..bbb..a
+a..b.bbb...a
+a..b.bb..b.a
+a..b.bbb...a
+a..b.bbb...a
+a..be.b...ta
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaa
+ap.b.......a
+a.b.aaaaaa.a
+abb..e..b..a
+a..aaaaab..a
+a.......bb.a
+a......b.b.a
+a.......b..a
+a.......bt.a
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaa
+ap.b.......a
+a.bbaaaaaa.a
+abb........a
+a..aaaaab..a
+a..aaaaabb.a
+a...aaab.b.a
+a....aaab.aa
+ae.....bbt.a
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaa
+ap.......e.a
+a..b.bbbbaaa
+a..bb..bb..a
+a...b.b.b..a
+a..bbbbbb..a
+a..bb..bb..a
+a..bb.bb...a
+a..bbb.bb.ta
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaa
+ap.b.bbbbbba
+abbb.b....ba
+a.b.bbbbbbba
+abb.b.b.bb.a
+abeb.b.b.bba
+abbbbb..b.ba
+ab.b..bb.b.a
+abb.bb.b..ta
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaa
+abtb.bb.bb.a
+a...b..b..ba
+a..a.a.aab.a
+a.a.bbbaa.ba
+abab..ba.b.a
+a.e...b.b..a
+abab...a.b.a
+abb.b.b.b.pa
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaa
+a..bb......a
+a.p..e.ab..a
+aa...b..b..a
+a..bbbaabbba
+a.bb.b.abbba
+a.b.a....t.a
+a..b.bbbbbba
+a.b.....a.aa
+aaaaaaaaaaaa`,
+  map`
+aaaaaaaaaaaaaaaa
+ap.............a
+a......e.......a
+a.abbaaaaaaaaaaa
+a.a..a...b..b.ba
+a.a..a.b.b..bb.a
+a.abbabb.aaaa..a
+a.a..ab.....a..a
+a.a..aaaa.t.a..a
+a.abb...baaaabba
+a.a.....b......a
+a.aaaaaaaaaaaaaa
+a..............a
+aaaaaaaaaaaaaaaa`,
+]
+
+const maxMoves = [
+  15, 15, 19, 18, 21, 19, 18, 22, 19, 17, 42
+]
+
+let curMoves = 0;
+
+const melody = tune`
+500: D4~500,
+500: D4~500 + A4^500,
+500: E4~500 + A4^500 + B4^500,
+500: E4~500 + B4^500,
+500: F4~500 + B4^500 + C5^500,
+500: F4~500 + C5^500 + D5^500,
+500: E4~500 + D5^500 + F5^500,
+500: E4~500,
+500: D4~500,
+500: D4~500,
+500: E4~500,
+500: E4~500 + B4^500,
+500: F4~500 + B4^500,
+500: F4~500 + B4^500 + C5^500,
+500: E4~500 + C5^500 + F5^500,
+500: E4~500,
+500: D4~500,
+500: D4~500,
+500: E4~500,
+500: E4~500 + B4^500,
+500: F4~500 + B4^500,
+500: F4~500 + B4^500 + A4^500,
+500: E4~500 + A4^500 + C4^500,
+500: E4~500,
+500: D4~500,
+500: D4~500,
+500: E4~500 + C5^500,
+500: E4~500 + B4^500 + C5^500,
+500: F4~500 + B4^500,
+500: F4~500 + A4^500 + C4^500,
+500: E4~500 + A4^500 + C4^500,
+500: E4~500`
+
+const playback = playTune(melody, Infinity)
+
+setMap(levels[level])
+
+setPushables({
+  [ player ]: [rock2]
+})
+
+addText((maxMoves[level]-curMoves).toString(), { 
+  x: 1,
+  y: 1,
+  color: color`3`
+})
+
+
+
+
+function findShortestPath(x, y) {
```

Comment:
Excellent work!

Reply:
Hmmm... after playing the game though, I am not sure if this functions as expected...  would that surprise you?  do you expect the enemy to follow the player w/ an optimal path?